api/src/main/java/uk/gov/dhsc/htbhf/dwp/service/HouseholdVerifier.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.dhsc.htbhf.dwp.service;
+
+import org.flywaydb.core.internal.util.StringUtils;
+import org.springframework.stereotype.Service;
+import uk.gov.dhsc.htbhf.dwp.entity.legacy.LegacyHousehold;
+import uk.gov.dhsc.htbhf.dwp.entity.uc.UCHousehold;
+import uk.gov.dhsc.htbhf.dwp.model.PersonDTO;
+
+@Service
+public class HouseholdVerifier {
+
+    /**
+     * Determines if a given household contains an adult matching a given person.
+     * DWP rules specify to only match on the first six characters of the address line 1 field.
+     * @param household household to check
+     * @param person person to check
+     * @return true the details of the household and the person match.
+     */
+    public Boolean detailsMatch(UCHousehold household, PersonDTO person) {
+        return household.getAdults().stream()
+                .anyMatch(adult ->
```

Comment:
I think this method and the one below need a bit of refactoring, the contents of the `anyMatch` can be refactored out into methods, and also the return statement in the next function for clarity.

Reply:
I'm not sure a method per equality check would read better than what's already there.

Reply:
Done.